src/sidebar/components/hooks/use-user-filter-options.js

```diff
@@ -0,0 +1,79 @@
+import { useMemo } from 'preact/hooks';
+
+import { useStoreProxy } from '../../store/use-store';
+import { username } from '../../util/account-id';
+
+/** @typedef {import('../../store/modules/filters').FilterOption} FilterOption */
+
+/**
+ * Generate a list of users for filtering annotations; update when set of
+ * annotations or filter state changes meaningfully.
+ *
+ * @return {FilterOption[]}
+ */
+export default function useUserFilterOptions() {
+  const store = useStoreProxy();
+  const annotations = store.allAnnotations();
+  const focusFilters = store.getFocusFilters();
+  const showDisplayNames = store.isFeatureEnabled('client_display_names');
+
+  return useMemo(() => {
+    // Determine unique users (authors) in annotation collection
+    const users = {};
+    annotations.forEach(annotation => {
+      // Display a username, not an account ID. If we're dealing with an
+      // account ID, we need to username() it
+      const _username = username(annotation.user)
+        ? username(annotation.user)
+        : annotation.user;
+      const displayValue =
+        showDisplayNames && annotation.user_info?.display_name
+          ? annotation.user_info.display_name
+          : _username;
+      users[annotation.user] = displayValue;
+    });
+
+    // If user-focus is configured (even if not applied) add a filter
+    // option for that user. Note that this always respects the display
+    // value, even if `client_display_names` feature flags is not enabled:
+    // this matches current implementation of focus mode.
+    if (focusFilters.user) {
+      // We need to see if the user indicated by the user-focus filter is
+      // already contained in `users`. This is complicated by the fact that
+      // the value of the user filter can be either an account ID or a username;
+      // we can't be sure...
+      const maybeUsername = focusFilters.user.value;
+      const alreadyPresent = Object.keys(users).find(probablyAccountId => {
```

Comment:
If my assumption about the validity of `annotation.user` above is correct, then the keys of `users` _will_ always be valid account IDs.

Reply:
A'ight, I'll make that assumption...